classifier/src/Lib.hs

```diff
@@ -363,68 +369,67 @@ inspectAttachments dataLayer ticketInfo attachments = do
 
     lastAttach      <- handleMaybe . safeHead . reverse . sort $ attachments
     att             <- handleMaybe =<< getAttachment lastAttach
-    inspectAttachment config ticketInfo att
+    inspectAttachment config ticketInfo att extractLogsFromZip extractIssuesFromLogs
   where
     handleMaybe :: Maybe a -> App a
     handleMaybe Nothing  = throwM $ AttachmentNotFound (tiId ticketInfo)
     handleMaybe (Just a) = return a
 
+-- This is to enable test cases on 'inspectAttachment'
+-- Perhaps we can add this to 'Config' ?
+type ExtractLogFileFunc = Int -> LByteString -> Either ZipFileExceptions [LogFile]
+type ExtractErrorCodeFunc m = [LogFile] -> Analysis -> m Analysis
+
+
 -- | Given number of file of inspect, knowledgebase and attachment,
 -- analyze the logs and return the results.
-inspectAttachment :: (MonadCatch m) => Config -> TicketInfo -> AttachmentContent -> m ZendeskResponse
-inspectAttachment Config{..} ticketInfo@TicketInfo{..} attachment = do
-
-    let analysisEnv = setupAnalysis cfgKnowledgebase
-    let eLogFiles = extractLogsFromZip cfgNumOfLogsToAnalyze (getAttachmentContent attachment)
-
-    case eLogFiles of
-        Left _ ->
-            -- Log file was corrupted
-            pure $ ZendeskResponse
-                { zrTicketId    = tiId
-                , zrComment     = prettyFormatLogReadError ticketInfo
-                , zrTags        = TicketTags [renderErrorCode SentLogCorrupted]
-                , zrIsPublic    = cfgIsCommentPublic
-                }
-
-        Right logFiles -> do
-            -- Log files maybe corrupted or issue was not found
-            tryAnalysisResult    <- try $ extractIssuesFromLogs logFiles analysisEnv
+inspectAttachment :: (MonadCatch m)
+                  => Config
+                  -> TicketInfo
+                  -> AttachmentContent
+                  -> ExtractLogFileFunc
+                  -> ExtractErrorCodeFunc m
+                  -> m ZendeskResponse
+inspectAttachment Config{..} ticketInfo attachment extractLogFileFunc extractIssueFunc =
+    flip catches [Handler handleZipFileException, Handler handleLogAnalysisException] $ do
```

Comment:
https://github.com/fpco/safe-exceptions/blob/master/src/Control/Exception/Safe.hs#L265

Reply:
Well, since I'm catching multiple exceptions, I'd need `handles` function which should be
`
``
handles :: (C.MonadCatch m, C.MonadThrow m) => [Handler m a] -> m a-> m a
handles handlers io = flip catches handlers io
```